### PR TITLE
Add missing build test dependency

### PR DESCRIPTION
**Pull request by ssbarnea**
Fixes: #2576

# Thanks for contribution

Please, make sure you address all the checklists (for details on how see
[development documentation](http://tox.readthedocs.org/en/latest/development.html#development))!

- [x] ran the linter to address style issues (`tox -e fix_lint`)
- [x] wrote descriptive pull request text
- [x] ensured there are test(s) validating the fix
- [x] added news fragment in `docs/changelog` folder
- [x] updated/extended the documentation

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -44,6 +44,7 @@ optional-dependencies.docs = [
   "towncrier>=22.8",
 ]
 optional-dependencies.testing = [
+  "build>=0.9.0",
```
Comment (gaborbernat) [inline pyproject.toml:47]: Let's make it `build[virtualenv]` and you can drop trailing .0-s

**Review by gaborbernat — CHANGES_REQUESTED** [review]

**Comment by mgorny** [comment]
Thank you!

**Comment by ssbarnea** [comment]
@jcristau I do not remember and do not plan to spend my free time investigating it. Feel free to check their changelog and see if there are good or not reasons for the bump.

**Comment by jcristau** [comment]
I did, and can't see any.
Too bad, thanks anyway.  I guess there was already an unexplained bump in https://github.com/tox-dev/tox/commit/01004234a48a834a8f814b2639b5d78fbc4462a6 so I'll just pin tox to <4.

**Pull request merged**